servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/Single.java

```diff
@@ -740,7 +740,30 @@ public final Completable flatMapCompletable(Function<? super T, ? extends Comple
      * all elements from {@code next} {@link Publisher}.
```

Comment:
This method should document that it is equivalent to concat(next, false) and explain that it will try to subscribe before the next item is needed.

I am also wondering how common it is for a Publisher to not allow concurrent subscribes. If it is at least somewhat common then perhaps the default should be true, just for safety. What are the execution or performance implications of deferring the subscribe? Would there be a downside to always deferring the subscribe? Could/should Publishers be typed with a marker interface to indicate that they don't support concurrent subscribes?

Reply:
When we discussed with @Scottmitch the approach offline we agreed to keep the existing behavior as-is in case users expect it and have the new behavior opt-in. The new operator is more complex and expensive. Maybe in future we will keep only new behavior and make it the default. But probably we will need to rework all other operators to behave in the same way (there are more `concat` variants for other sources).